src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -311,5 +345,32 @@ private Uri CreateFullEndpointUri(Uri endPoint)
                 }
             }
         }
+
+        private IDictionary<string, string> ParseChallengeData(HttpResponse response)
+        {
+            IDictionary<string, string> data = new Dictionary<string, string>();
+            string wwwAuthenticate = response.Headers.GetValues(WwwAuthenticateHeader).FirstOrDefault();
```

Comment:
Use .Single if you are sure there is only one value. Otherwise, extract the right value using .Where